peer-pod-controller/main.go

```diff
@@ -84,6 +85,13 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "PeerPodConfig")
 		os.Exit(1)
 	}
+	if err = (&controllers.PeerPodReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "PeerPod")
+		os.Exit(1)
+	}
```

Comment:
As mentioned above, if we can specify a cloud provider name as a command line argument, we can create a Provider object here, and pass it in this struct.